src/camera.rs

```diff
@@ -432,3 +266,433 @@ impl Default for capi::CameraProperties_camera_properties_shape_s {
         }
     }
 }
+
+/// CameraCapabilities::OffsetShapeCapabilities
+#[pyclass]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OffsetShapeCapabilities {
+    #[pyo3(get, set)]
+    x: Property,
+
+    #[pyo3(get, set)]
+    y: Property,
+}
+
+impl_plain_old_dict!(OffsetShapeCapabilities);
+
+impl Default for OffsetShapeCapabilities {
+    fn default() -> Self {
+        Self {
+            x: Property::default(),
+            y: Property::default(),
+        }
+    }
+}
+
+impl TryFrom<capi::CameraPropertyMetadata_camera_properties_metadata_offset_s> for OffsetShapeCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::CameraPropertyMetadata_camera_properties_metadata_offset_s,
+    ) -> Result<Self, Self::Error> {
+        Ok(Python::with_gil(|_| -> PyResult<_> {
+            Ok(Self {
+                x: value.x.try_into()?,
+                y: value.y.try_into()?,
+            })
+        })?)
+    }
+}
+
+impl TryFrom<capi::CameraPropertyMetadata_camera_properties_metadata_shape_s> for OffsetShapeCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::CameraPropertyMetadata_camera_properties_metadata_shape_s,
+    ) -> Result<Self, Self::Error> {
+        Ok(Python::with_gil(|_| -> PyResult<_> {
+            Ok(Self {
+                x: value.x.try_into()?,
+                y: value.y.try_into()?,
+            })
+        })?)
+    }
+}
+
+/// CameraCapabilities::DigitalLineCapabilities
+#[pyclass]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DigitalLineCapabilities {
+    #[pyo3(get, set)]
+    line_count: u8,
+
+    #[pyo3(get, set)]
+    names: [String; 8],
+}
+
+impl_plain_old_dict!(DigitalLineCapabilities);
+
+impl Default for DigitalLineCapabilities {
+    fn default() -> Self {
+        Self {
+            line_count: Default::default(),
+            names: Default::default(),
+        }
+    }
+}
+
+impl TryFrom<capi::CameraPropertyMetadata_CameraPropertyMetadataDigitalLineMetadata> for DigitalLineCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::CameraPropertyMetadata_CameraPropertyMetadataDigitalLineMetadata,
+    ) -> Result<Self, Self::Error> {
+        Ok(Python::with_gil(|_| -> PyResult<_> {
+            let mut names: [String; 8] = Default::default();
+            for (i, name) in value.names.iter().enumerate() {
+                let name = unsafe { CStr::from_ptr(name.as_ptr()) }.to_str()?.to_owned();
+                names[i] = name;
+            }
+            Ok(Self {
+                line_count: value.line_count,
+                names,
+            })
+        })?)
+    }
+}
+
+/// CameraCapabilities::TriggerCapabilities
+#[pyclass]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TriggerCapabilities {
+    #[pyo3(get, set)]
+    acquisition_start: (u8, u8),
```

Comment:
The first element is the 8bit-field that indicates which digital lines can be used for the acquisition_start input trigger, right? And the second is the same for output triggers/lines?

If so, I don't love this reflection, because it's not very explicit. Is there a similar one elsewhere in Acquire? Why not just reflect Acquire's C API (i.e. `TriggerCapabilities.acquisition_start.input`)?

Reply:
No real reason, just one less struct to have to reflect. Another easy fix.